src/analyzer.ts

```diff
@@ -276,34 +283,47 @@ export class Analyzer {
       firstScannedFeature.applyHtmlComment(maybeAttachedComment);
     }
 
-    const scannedDependencies: ScannedFeature[] = scannedFeatures.filter(
-        (e) =>
-            e instanceof ScannedInlineDocument || e instanceof ScannedImport);
+    const scannedDocument =
+        new ScannedDocument(document, scannedFeatures, warnings);
+
+    if (!scannedDocument.isInline) {
+      this._scannedDocuments.set(scannedDocument.url, scannedDocument);
+    }
+    return scannedDocument;
+  }
+
+  /**
+   * Scan all the dependencies of the given scanned document.
+   *
+   * This must be called exactly once per scanned document, as we mutate
+   * the given scannedDocument by adding warnings.
+   */
+  private async _scanDependencies(scannedDocument: ScannedDocument) {
```

Comment:
nit: can you add a return type here? it's hard to tell what this function is meant to return